ngi_reports/common/ign_sample_report.py

```diff
@@ -400,17 +412,24 @@ def parse_template(self, template):
 
         output_mds = {}
 
-        self.LOG.info('Processing reports')
+        self.LOG.debug('Processing reports')
+
+        # Check that we have mandatory project-level fields
+        if not self.check_project_fields():
+            self.LOG.error("Missing mandatory project-level fields, exiting..")
+            raise KeyError
+
         # Go through each sample making the report
         for sample_id, sample in sorted(self.samples.iteritems()):
 
+            self.LOG.debug("Generating report for {}".format(sample_id))
+
             # Make the file basename
             report_fn = sample_id + '_ign_sample_report'
             output_bn = os.path.realpath(os.path.join(self.working_dir, self.report_dir, report_fn))
 
-            # check that we have everythin
-            if not self.check_fields():
-                self.LOG.error("Some mandatory fields were missing for sample {} - skipping".format(sample_id))
+            # check that we have everything (errors thrown by def)
+            if not self.check_sample_fields(sample_id):
```

Comment:
so I guess that in this case you're not aborting like in line 420 (when `check_project_fields` is called) because you don't wan to stop for all samples, right? And you're not logging anything here because the error is logged in the `check_sample_fields` method, right?

I'm just making sure I got it :-p